ivy/functional/frontends/jax/numpy/creation.py

```diff
@@ -243,3 +243,9 @@ def compress(condition, a, *, axis=None, out=None):
         )
     arr = arr[: condition_arr.shape[0]]
     return ivy.moveaxis(arr[condition_arr], 0, axis)
+
+
+@inputs_to_ivy_arrays
+def iterable(y):
+    return hasattr(y, "__iter__") or y.ndim > 0
```

Comment:
You'll use `and` because `0-D` arrays are not iterable.

```python
import numpy as np
print(np.iterable(2)) # False
```

Reply:
The hasattr for a 0-D array will return a False and thereby the iterable function will return a false.
https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.iterable.html says either a sequence or has an iterator method.

Reply:
It should be `and` because we convert the input to `ivy.Array` instance and the `ivy.Array` class has `__iter__` method which will always be true even for `0-D` arrays

Reply:
Oh, so there's no requirement to assert if the object has an iter method? Since all input objects are converted to an instance of ivy.array and will have an iter method

Reply:
The reason to have this assertion is that they're objects that are scalars for example if we remove `hasattr` assertion the function will fail for scalar inputs because we will be checking the `ndim` of scalar inputs as the [`inputs_to_ivy_arrays`](https://github.com/unifyai/ivy/blob/e3797979e4298ce6b0cfb16bba7fa29ac9605965/ivy/functional/frontends/numpy/func_wrapper.py#LL326C1-L326C1) wrapper doesn't convert scalars to `ivy.Array` instances.

Reply:
we want the function to return False for scalar inputs and it does that

Reply:
I explained why we need that `hasattr` assertion now that the tests are passing and I'm about to merge the PR.
Unless there is any other suggestion you want to add😊

Reply:
Okay, thank you very much for being patient and answering all my queries :)
Do you have suggestions on what issue to pick next? And in general to have a better understanding of the codebase.

Reply:
🤞ery task is an opportunity to explore the codebase I am sure you'll learn a lot with any item you pick🙂
All the best